src/ck-ldb/TreeLB.C

```diff
@@ -198,6 +202,21 @@ void TreeLB::configure(json& config)
 #endif
 }
 
+void TreeLB::pup(PUP::er& p)
+{
+  std::string configString;
+  if (p.isPacking())
+  {
+    configString = config.dump();
+  }
+  p | configString;
```

Comment:
Isn't this enforcing that you have the same LB config before and after restart? Shouldn't I be able to restart with whatever LB config I want, even if I didn't use any LB before restart?

Reply:
Potentially, that's where this gets a little tricky. On one hand, it perhaps makes sense to save the LBs used in the original run so that they'll be automatically used at restart, but on the other, they're not intrinsically a part of the application and one may want to restart with another LB, as you're saying.

Reply:
Yeah, I very strongly fall into camp #2. The load balancer is like the PE count. Not part of the application, part of the run. And when I rerun the application, I should be able to configure the run how I want w.r.t. number of PEs, whether I'm using LB, etc.

But, I think either way, if this fixes the crash, then we can merge this and discuss in core or something how we actually think this should work. The important thing now is fixing the bug.